Broken build with OMP enabled

WFA2-lib is not building for me with cmake version 3.25.2 when OpenMP is enabled.

It looks like https://github.com/smarco/WFA2-lib/pull/64 illegally used a mix of plain and keyword-based CMake syntax for linking libraries: https://stackoverflow.com/questions/59522267/cmake-rejects-a-second-target-link-libraries-talking-about-keyword-vs-plain

Reproduction:

Build with OpenMP: 
```
mkdir build &&  cd build
cmake -DOPENMP=ON ..
cmake --build .
```
Fail:

```
CMake Error at CMakeLists.txt:176 (target_link_libraries):
  The keyword signature for target_link_libraries has already been used with
  the target "wfa2cpp_static".  All uses of target_link_libraries with a
  target must be either all-keyword or all-plain.

  The uses of the keyword signature are here:

   * CMakeLists.txt:171 (target_link_libraries)



CMake Error at CMakeLists.txt:177 (target_link_libraries):
  The keyword signature for target_link_libraries has already been used with
  the target "wfa2cpp".  All uses of target_link_libraries with a target must
  be either all-keyword or all-plain.

  The uses of the keyword signature are here:

   * CMakeLists.txt:170 (target_link_libraries)
```

Solutions: Specify the scope of how OpenMP is linked to WFA2-lib. I don't see any reason to publicly expose this, so `PRIVATE` seems like the default option.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -151,8 +151,8 @@ add_library(wfa2::wfa2 ALIAS wfa2)
 add_library(wfa2::wfa2_static ALIAS wfa2_static)
 
 if(OPENMP)
-  target_link_libraries(wfa2_static OpenMP::OpenMP_C)
-  target_link_libraries(wfa2 OpenMP::OpenMP_C)
+  target_link_libraries(wfa2_static PRIVATE OpenMP::OpenMP_C)
+  target_link_libraries(wfa2 PRIVATE OpenMP::OpenMP_C)
 endif(OPENMP)
 
 # ---- C++ binding library
@@ -173,8 +173,8 @@ add_library(wfa2::wfa2cpp ALIAS wfa2cpp)
 add_library(wfa2::wfa2cpp_static ALIAS wfa2cpp_static)
 
 if(OPENMP)
-  target_link_libraries(wfa2cpp_static OpenMP::OpenMP_CXX)
-  target_link_libraries(wfa2cpp OpenMP::OpenMP_CXX)
+  target_link_libraries(wfa2cpp_static PRIVATE OpenMP::OpenMP_CXX)
+  target_link_libraries(wfa2cpp PRIVATE OpenMP::OpenMP_CXX)
 endif(OPENMP)
 
 # ---- Get version

